src/app/js/formats/bubbleChart/BubbleAdmin.js

```diff
@@ -50,20 +51,32 @@ class BubbleAdmin extends Component {
         args: defaultArgs,
     };
 
+    constructor(props) {
+        super(props);
+        this.handleColorsChange = this.handleColorsChange.bind(this);
+        this.state = {
+            colors:
+                this.props.args.colors != null
+                    ? this.props.args.colors
+                    : colorUtils.MULTICHROMATIC_DEFAULT_COLORSET,
+        };
+    }
+
     setParams = params => updateAdminArgs('params', params, this.props);
 
     setDiameter = (_, diameter) => {
         updateAdminArgs('diameter', diameter, this.props);
     };
 
-    setColors = (_, colors) => {
+    handleColorsChange(colors) {
         updateAdminArgs('colors', colors, this.props);
-    };
+        this.setState({ colors });
```

Comment:
Why save the `colors` in the state?
That ColorPickerParamsAdmin component should deal with `colors`, shouldn't it?

Reply:
> [...] we want these two inputs to be in sync with each other. [...]
In React, sharing state is accomplished by moving it up to the closest common ancestor of the components that need it. This is called “lifting state up”. We will remove the local state from the [Child] and move it into the [Parent] instead.
([React docs](https://reactjs.org/docs/lifting-state-up.html#lifting-state-up))

So I guess it is a classic pattern when you want to translate data from a Child Component to his Parent.

Reply:
Just made a change to stop using `props.colors` in the Component : I managed to use it effectively as read only data.
This way, all the changes are inside of the Component, with `state.colors`, and this `setState` is now useless in the parent ! 😄